pbft/pbft.cpp

```diff
@@ -710,6 +724,36 @@ pbft::do_committed(const std::shared_ptr<pbft_operation>& op)
     }
 }
 
+void
+pbft::handle_new_config_timeout(const boost::system::error_code& ec)
+{
+    if (ec == boost::asio::error::operation_aborted)
+    {
+        return;
+    }
+
+    if (ec)
+    {
+        LOG(error) << "handle_new_config_timeout error: " << ec.message();
+        return;
+    }
+
+    // send viewchange with new config
+    // TODO: should the current view be invalidated?
+    auto config = this->configurations.newest();
+    if (config)
+    {
+        this->view_is_valid = false;
```

Comment:
I don't like duplicating this logic; can we invoke the method that the failure detector invokes? (perhaps renaming it)

Reply:
done